### PR TITLE
Update DG for reminders feature

### PR DESCRIPTION
**Pull request by cashewnade**
this will close AY2122S2-CS2103T-T17-3#214

### CODE REVIEW

**Review by pyk595 — APPROVED** [review]
LGTM

**Review by ngjunkang — CHANGES_REQUESTED** [review]
Hi, many of the activation bar does not end at the right place, for example, ParserUtil in AddReminderCommandSequenceDiagram. The activation bar should be separated. one for description and other for date.

File: docs/diagrams/DeleteReminderFromReminderListSeqDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+create ReminderListCopy
+ReminderList -> ReminderListCopy : getCopy()
+activate ReminderListCopy
+
+create PriorityQueueCopy
+ReminderListCopy -> PriorityQueueCopy : new PriorityQueue(this.reminderPriorityQueue)
+activate PriorityQueueCopy
+
+PriorityQueueCopy --> ReminderListCopy : reminderPriorityQueueCopy
```
Comment (pyk595) [inline docs/diagrams/DeleteReminderFromReminderListSeqDiagram.puml:39]: Move line 39 and 40 to line 56

File: docs/diagrams/DeleteReminderFromReminderListSeqDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+deactivate PriorityQueueCopy
+
+ReminderListCopy --> ReminderList : ReminderListCopy
+deactivate ReminderListCopy
+
+ReminderList -> PriorityQueueCopy : reminderPriorityQueue.remove(reminder)
```
Comment (pyk595) [inline docs/diagrams/DeleteReminderFromReminderListSeqDiagram.puml:45]: `ReminderListCopy->PriorityQueueCopy: remove(reminder)`

File: docs/diagrams/DeleteReminderFromReminderListSeqDiagram.puml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+activate PriorityQueueCopy
+
+PriorityQueueCopy -> PriorityQueueCopy : remove(reminder)
+activate PriorityQueueCopy
+deactivate PriorityQueueCopy
+
+PriorityQueueCopy --> ReminderList : reminderPriorityQueueCopy
```
Comment (pyk595) [inline docs/diagrams/DeleteReminderFromReminderListSeqDiagram.puml:52]: `PriorityQueueCopy-->ReminderListCopy`...the other stuff

File: docs/diagrams/DeleteReminderFromReminderListSeqDiagram.puml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+PriorityQueueCopy --> ReminderListCopy : reminderPriorityQueueCopy
+deactivate PriorityQueueCopy
+
+ReminderListCopy --> ReminderList : ReminderListCopy
+deactivate ReminderListCopy
+
+ReminderList -> PriorityQueueCopy : reminderPriorityQueue.remove(reminder)
+activate PriorityQueueCopy
+
+PriorityQueueCopy -> PriorityQueueCopy : remove(reminder)
```
Comment (pyk595) [inline docs/diagrams/DeleteReminderFromReminderListSeqDiagram.puml:48]: line 48 to 50 can delete

File: docs/diagrams/AddReminderCommandParserSeqDiagram.puml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+deactivate ReminderDate
+
+ParserUtil --> AddReminderCommandParser : ReminderDate
+deactivate ParserUtil
+
+create Reminder
+AddReminderCommandParser -> Reminder : new Reminder(reminderDescription, reminderDate)
+activate Reminder
+
+Reminder --> AddReminderCommandParser : Reminder
```
Comment (pyk595) [inline docs/diagrams/AddReminderCommandParserSeqDiagram.puml:55]: `Reminder --> AddReminderCommandParser: toAdd`

File: docs/diagrams/AddReminderCommandParserSeqDiagram.puml [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+activate AddReminderCommandParser
+AddReminderCommandParser -> ParserUtil : parseReminderDescription(reminderDescription)
+activate ParserUtil
+
+create ReminderDescription
+ParserUtil -> ReminderDescription : new ReminderDescription(reminderDescription)
+activate ReminderDescription
+
+ReminderDescription --> ParserUtil : ReminderDescription
```
Comment (pyk595) [inline docs/diagrams/AddReminderCommandParserSeqDiagram.puml:32]: `ReminderDescription --> ParserUtil: reminderDescription`

**Review by pyk595 — CHANGES_REQUESTED** [review]
Some things to fix

File: docs/diagrams/AddReminderCommandParserSeqDiagram.puml [hunk 7]
```diff
@@ -29,7 +29,7 @@ create ReminderDescription
 ParserUtil -> ReminderDescription : new ReminderDescription(reminderDescription)
 activate ReminderDescription
 
-ReminderDescription --> ParserUtil : ReminderDescription
+ReminderDescription --> ParserUtil: reminderDescription
```
Comment (pyk595) [inline docs/diagrams/AddReminderCommandParserSeqDiagram.puml:32]: II gave you some comments in telegram, you can refer to those and make some small edits, thanks!

**Review by pyk595 — CHANGES_REQUESTED** [review]

**Review by pyk595 — APPROVED** [review]
LGTM

**Review by ngjunkang — APPROVED** [review]

**Pull request merged**